validator.go

```diff
@@ -179,3 +189,20 @@ func WithAllowUnknownFields() ValidatorOption {
 		c.allowUnknownFields = true
 	}
 }
+
+type validationConfig struct {
+	failFast bool
+	filter   Filter
+}
+
+// A ValidationOption specifies per-validation configuration. See the individual
+// options for their defaults and effects.
+type ValidationOption func(*validationConfig)
+
+// WithFilter specifies a filter to use for this validation. A filter can
+// control which fields are evaluated by the validator.
+func WithFilter(filter Filter) ValidationOption {
+	return func(c *validationConfig) {
+		c.filter = filter
```

Comment:
```suggestion
		c.filter = cmp.Or(filter, nopFilter{})
```

Reply:
Can't use `cmp.Or` yet, since our module version is go 1.21.1 and this was added in go 1.22. Now that go 1.24 is out though maybe it's time to bump that number. Either way, for now I just did the simpler thing and made a conditional for this.